src/itkVkGlobalConfiguration.cxx

```diff
@@ -0,0 +1,80 @@
+/*=========================================================================
+ *
+ *  Copyright NumFOCUS
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0.txt
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ *
+ *=========================================================================*/
+#include "itkVkGlobalConfiguration.h"
+
+#include <mutex>
+#include "itkSingleton.h"
+
+namespace itk
+{
+struct VkGlobalConfigurationGlobals
+{
+  VkGlobalConfigurationGlobals()
+    : m_Instance(nullptr){};
+
+  VkGlobalConfiguration::Pointer m_Instance;
```

Comment:
We could replace the `m_Instance(nullptr)` by moving it to the the declaration line:
```cpp
VkGlobalConfiguration::Pointer m_Instance{ nullptr };
```